File: algorithms/1/가장 긴 팰린드롬/가장 긴 팰린드롬.py
def palindrome(s:str, length:int) -> bool:
    for ele in range(len(s)-length+1):
        a = s[ele:ele+length]
        if a == a[::-1]:
            return True
    return False


def solution(s:str)-> int:
    n = len(s)
    if s == s[::-1]:
        return n
    else:
        for i in range(len(s), 0, -1):
            if palindrome(s, i):
                return i
        return 1

File: algorithms/1/가장 긴 팰린드롬/가장 긴 팰린드롬.cpp
#include <bits/stdc++.h>

using namespace std;

bool palindrome(string s, int length)
{
    for (int i = 0; i < s.size() - length + 1; i++)
    {
        string temp_a(s.begin() + i, s.begin() + i + length);
        string temp_b = temp_a;
        reverse(temp_b.begin(), temp_b.end());
        if (temp_a == temp_b)
        {
            return true;
        }
    }
    return false;
}

int solution(string s)
{
    string rs = s;
    reverse(rs.begin(), rs.end());
    int n = s.size();
    if (s == rs)
    {
        return s.size();
    }
    else
    {
        for (int i = n; i >= 2; i--)
        {
            if (palindrome(s, i))
                return i;
        }

        return 1;
    }
}